### ISSUE TITLE
Add support for django-rest-framework-simplejwt as JWT authentication backend

### ISSUE THREAD

**Issue by JoelGoh92** [Tivix/django-rest-auth#430]
According to an [issue #5838 on DRF repository](https://github.com/encode/django-rest-framework/issues/5838) and a [PR](https://github.com/encode/django-rest-framework/pull/5251) raised on the DRF repository, it seems [django-rest-framework-simplejwt](https://github.com/davesque/django-rest-framework-simplejwt/) provides a much better implementation of JWT over [django-rest-framework-jwt](https://github.com/GetBlimp/django-rest-framework-jwt), with regards to security and use of JWT as an authentication mechanism. Hence, I'm curious if you guys will be planning to include the simplejwt package as a supported JWT backend? Thanks

**Comment by Allan-Nava** [comment]
Yes I need to implement the viewsets.ModelViewSet with authentication_classes for specific action. For example update or create new instance.

Thank's in advance.
Allan

**Comment by JoelGoh92** [comment]
Thanks for the reply, looking forward to this!

**Comment by Allan-Nava** [comment]
So I don't use https://github.com/GetBlimp/django-rest-framework-jwt but I have to use django-rest-framework-simplejwt? Correct?

But is possible for specific action like update or create ?

**Comment by Allan-Nava** [comment]
@JoelGoh92 But I have the JWT thanks to a keycloak

**Comment by JoelGoh92** [comment]
@Allan-Nava currently for our project, we're not using [the rest-auth jwt setup defined in the docs](http://django-rest-auth.readthedocs.io/en/latest/installation.html#jwt-support-optional). Instead, after setting up simple-jwt with [the steps in the docs](https://github.com/davesque/django-rest-framework-simplejwt#installation), and configuring the [settings](https://github.com/davesque/django-rest-framework-simplejwt#settings), we then wrote our own JWTSocialLoginView, JWTLoginView and JWTLogoutView, which were fairly simple, and used them in the required areas. 

However I still feel that it is better if these changes were to be supported and maintained by rest-auth instead.

**Comment by Allan-Nava** [comment]
So is not possible in modelviewsets?

Inviato da iPhone

**Comment by JoelGoh92** [comment]
@Allan-Nava It may be possible, but I would not recommend the model viewsets though, because these defined view classes will be used in a similar way as to how the current LoginView/SocialLoginView/LogoutView provided by rest-auth are used, and we only need to override the rest-auth implementations' required methods for the JWT token response and usage flow.

The JWT views I mentioned previously are inheriting from the respective classes rest-auth provides, e.g. 
```python
class JWTSocialLoginView (SocialLoginView):
     def get_response (self):
     # override method(s) for JWT implementation
     # get JWT token via the simple-jwt package
     # return the generated token in the response
```

This is because such a class will be reused in a similar manner, e.g. the JWTSocialLoginView defined will be used similarly to how the SocialLoginView provided by rest-auth for OAuth apps is used, other than returning the JWT token(s) in tbe response

**Comment by Allan-Nava** [comment]
But is possibile to use the permission class only for specific action like update or delete?

**Comment by JoelGoh92** [comment]
@Allan-Nava I think the use of JWT here is more for [authentication](http://www.django-rest-framework.org/api-guide/authentication/#json-web-token-authentication). With DRF, you can set it up pretty easily, e.g. as a default authentication class.

If you're looking to implement permissions control, I would advise you to look at [DRF's docs on permissions control](http://www.django-rest-framework.org/api-guide/permissions/). This should not have anything to do with whether you're using jwt as an auth mechanism.

**Comment by lukeburden** [comment]
While `django-rest-framework-simplejwt` is a newer library with fewer contributors, it is more recently maintained, has higher code quality than `django-rest-framework-jwt`, allows for the more typical JWT implementation (refresh and access tokens are different) and has extras such as refresh token blacklisting built in.

**Comment by whwkong** [comment]
It would be good to have `django-rest-framework-simplejwt` support.  
It doesn't look like `django-rest-framework-jwt` is being actively maintained.  Last commit was Oct, 2017.

**Comment by aaronrosenberg** [comment]
Just throwing my support behind django-rest-framework-simplejwt.   Would be a huge benefit and streamline the whole authentication workflow,

**Comment by robypomoni** [comment]
+1 on this. Please add support to django-rest-framework-simplejwt

**Comment by superandrew** [comment]
+1, I would also love to have support for simplejwt!

**Comment by sm087** [comment]
+1 for the support of simplejwt

**Comment by alexferrari88** [comment]
+1 for the support of simplejwt

**Comment by NidalM** [comment]
As per the latest status update on the `django-rest-framework-jwt` page, this repo is no longer being actively maintained.
https://github.com/jpadilla/django-rest-framework-jwt/issues/484

**Comment by slystone** [comment]
+1 for the support of simplejwt!

**Comment by slystone** [comment]
@JoelGoh92 can you please be more specific about the solution of the problem? I'm not so confident with all the LoginView/SocialLoginView/LogoutView overriding thing you suggested

**Comment by sundeepdev** [comment]
This issue is more than year old. Does someone has a PR for this or suggestion on how to implement it? I just don't want to reinvent the wheel and can work on some other issue.

**Comment by NidalM** [comment]
How to use django-rest-framework-simplejwt as auth backend for django-rest-auth. Note: your needs may differ slightly based on how you implemented rest-auth.

Check your urls.py for where you route the rest-auth endpoints:
```
urlpatterns = [
    ...
    url(r'^rest-auth/', include('rest_auth.urls')),
    ...
]
```

You need to override the /rest-auth/login/ path with the login view from django-rest-framework-simplejwt by including it before the rest-auth urls. Here's how:

```
from rest_framework_simplejwt.views import TokenObtainPairView
...
urlpatterns = [
    ...
    url(r'^rest-auth/login/$', TokenObtainPairView.as_view(), name='rest_login'),
    url(r'^rest-auth/', include('rest_auth.urls')),
    ...
]
...
```

This'll get you started but there may be other routes you may need to override/create (e.g. logout). If you want the token login response to have non-default data, then you can override TokenObtainPairView with a custom serializer. Finally, because you're migrating from drf-jwt, you may need to change the auth header in settings.py to:
```
SIMPLE_JWT = {
    ...
    'AUTH_HEADER_TYPES': ('JWT', 'Bearer'),
}
```

The above worked for me, but was specific to my use case.

**Comment by sundeepdev** [comment]
Thanks @NidalM. This is helpful. However, I'm actually going to use Simple-JWT for both email login as well as social login and I was trying to avoid writing my own end points for all the features supported by this library.
So if I override login by completely writing my own end point, wouldn't I need to write the end points for everything like Registration, Forgot Password, Social-Auth for every provider etc.
I was hoping that there should be a way to hook in simple-jwt instead of drw-jwt by overriding a common method of JWT token generation and the rest could remain the same.
I'm not sure if there's no way to do that and we have to either write all the end points or change the complete implementation of the JWT token part of this library.

**Comment by jamesdvance** [comment]
+ 1 for simple_jwt! Commenting to follow this chain

**Comment by birgert** [comment]
As many others requested... Out of the box simplejwt support would make things so much easier. It's also the suggested JWT package by DRF.

**Comment by ankurpandeyvns** [comment]
+1 for simple JWT! It's an awesome library!

**Comment by ankurpandeyvns** [comment]
https://github.com/ankurpandeyvns/django-rest-auth/commit/794dac308da75537d2af15174b8644c0163a2b5a

This may be helpful for those who are looking only for JWT Logins.

**Comment by MIRAMAXED** [comment]
+1 for the support of simplejwt

**Comment by newbro** [comment]
+1 please, this library is extremely useful however the Django  token authentication is simply too insecure for my taste.

**Comment by ankurpandeyvns** [comment]
> +1 please, this library is extremely useful however the Django token authentication is simply too insecure for my taste.

https://github.com/ankurpandeyvns/django-rest-auth/commit/794dac308da75537d2af15174b8644c0163a2b5a


Try this one

**Comment by iamcb** [comment]
+1 for simple_JWT

**Comment by de-don** [comment]
+1

**Comment by vitzaoral** [comment]
+1

**Comment by bekaryukovmv** [comment]
+ 1 simple_JWT

**Comment by Dedaldino3D** [comment]
+1 simpleJWT

**Comment by quank123wip** [comment]
+1 simpleJWT

**Comment by rodrigondec** [comment]
+1 simpleJWT

**Comment by kikanny** [comment]
+1 simpleJWT pls!

**Comment by vThaian** [comment]
+1 for simpleJWT

**Comment by birgert** [comment]
Django-rest-auth was forked to  [dj-rest-auth](https://github.com/jazzband/dj-rest-auth). See #568 for some info. Lets hope this one becomes the new go to, it seems promising so far.

For all the people requesting simpleJWT, there's a pull request (ready to merge) in the new repo to add simplejwt support. https://github.com/jazzband/dj-rest-auth/pull/3

**Comment by pickyuptruck** [comment]
+1 simple JWT

**Comment by gaara4896** [comment]
+1 simple JWT

**Comment by kaniak274** [comment]
+1 simple JWT

**Comment by Issen007** [comment]
+1 simpleJWT
